protected/humhub/modules/file/models/File.php

```diff
@@ -159,6 +159,10 @@ public function canDelete($userId = null)
             return $object->content->canEdit($userId);
         }
 
+        if($object !== null && $object instanceof ActiveRecord){
+            return $object->canDelete();
```

Comment:
why do we use `canDelete` but `canEdit` in line 159?

Reply:
`canEdit` is used when user can edit the whole object when the object is of type Content or ContentAddon, but in our case our object is type of ActiveRecord for which we have related files but you're right in on point we should rename the method to be `CanDeleteFile()` to be more precise.

Reply:
don't fully understand why it is different? why can can the content be deleted if the user `canEdit()` it?

Reply:
It's different because `Funding Model` is not a space stream record that will be shown in the stream when entering the space, that why it doesn't have content to be edited and that's why its type is ActiveRecord and not ContentActiveRecord or ContentAddonActiveRecord, in other words since there's not content to be  edited we've only files that can be deleted from that came the method name, `canDeleteFile()` and not `canEdit()` .